src/App.js

```diff
@@ -1,109 +1,166 @@
+            />
+          }
+        ></Route>
+        <Route
+          path="/home/mobility"
+          element={
+            <CategoryMobilitiy
+              energyConsumptionHistory={energyConsumptionHistory}
+              dailyTotalBudget={dailyTotalBudget}
+              totalConsumption={totalConsumption}
+              setActiveChart={setActiveChart}
+              activeChart={activeChart}
+              messageText={messageText}
+              setEnergyConsumptionHistory={setEnergyConsumptionHistory}
+              updateEnergyConsumption={updateEnergyConsumption3}
+            />
+          }
+        ></Route>
+      </Routes>
+      <Footer />
     </>
   );
 
-  function updateChart() {
-    const chartInputData = {
-      labels: energyConsumptionHistory.reverse().map(input =>
-        input.date.toLocaleDateString('en-GB', {
-          year: 'numeric',
-          month: '2-digit',
-          day: '2-digit',
-        })
-      ),
-      datasets: [
-        {
-          label: 'Consumption Data',
-          data: energyConsumptionHistory.map(input => input.increase),
-          borderColor: energyConsumptionHistory.reverse()[0].increase > dailyTotalBudget ? 'red' : 'green',
-          borderWidth: '2',
-        },
-      ],
-    };
-    return chartInputData;
-  }
-
-  function updateEnergyConsumption(inputEnergyConsumptionValue) {
-    const newInput = {
+  function updateEnergyConsumption1(inputEnergyConsumptionValue) {
+    const newInput1 = {
       id: nanoid(),
       date: new Date(),
       value: Number(inputEnergyConsumptionValue),
-      increase: inputEnergyConsumptionValue - energyConsumptionHistory[0].value,
+      increase: inputEnergyConsumptionValue - energyConsumptionHistory.electric[0].value,
     };
 
-    if (newInput.value >= energyConsumptionHistory[0].value) {
-      setEnergyConsumptionHistory([newInput, ...energyConsumptionHistory]);
-      if (oldInputLength !== energyConsumptionHistory.length) {
+    if (newInput1.value >= energyConsumptionHistory.electric[0].value) {
+      setEnergyConsumptionHistory({
+        electric: [newInput1, ...energyConsumptionHistory.electric],
+        heating: [...energyConsumptionHistory.heating],
+        mobility: [...energyConsumptionHistory.mobility],
+      });
+      if (oldInputElectricLength !== energyConsumptionHistory.electric.length) {
         setMessageText(messages.success);
       } else {
         setMessageText(messages.dataNeeded);
       }
-      if (energyConsumptionHistory[0].increase > dailyTotalBudget) {
+      if (energyConsumptionHistory.electric[0].increase > dailyTotalBudget) {
         setMessageText(messages[2].text);
       }
     } else {
-      setMessageText('please input > or = ' + energyConsumptionHistory[0].value);
+      setMessageText('please input > or = ' + energyConsumptionHistory.electric[0].value);
     }
   }
+
+  function updateEnergyConsumption2(inputEnergyConsumptionValue) {
```

Comment:
I recommend that you define a function in such a way that you can use it multiple times.

Reply:
Thanks for your Comment: Can you give me a hint how can I solve this in a simple way? - I do not want to lose too much time;